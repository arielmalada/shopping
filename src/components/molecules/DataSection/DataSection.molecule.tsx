//#region IMPORTS
import { Typography } from "@material-tailwind/react";
import React, { useContext } from "react";
import { ProductContext } from "../../../contexts/ProductProvider";

//#endregion



//#region MAIN COMPONENTS
const DataSection: React.FC = () => {
  const productValue = useContext(ProductContext);
  const isPriceReduced = (productValue?.discountPercentage || 0) > 0;
  const finalPrice =
    isPriceReduced && productValue
      ? productValue.price -
        Math.round(productValue.price * productValue.discountPercentage * 0.01)
      : productValue?.price;
  return (
    <div className="col-span-12 space-y-3 lg:col-span-8">
      <Typography variant="h2">{productValue?.title}</Typography>
      <div>
        <div className="flex space-x-2">
          {isPriceReduced && (
            <div className="text-xs text-red-700 ">
              <div>Deal</div>
              <div className="text-sm">{productValue?.discountPercentage} %</div>
            </div>
          )}
          <div>$ {finalPrice}</div>
        </div>
        {isPriceReduced && (
          <div className="flex space-x-1 text-xs">
            <span>was</span>
            <span className="line-through">$ {productValue?.price}</span>
          </div>
        )}
      </div>
      <div>{productValue?.description}</div>
    </div>
  );
};
//#endregion

export default DataSection;