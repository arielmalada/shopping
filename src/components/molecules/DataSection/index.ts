export {default} from './DataSection.molecule'