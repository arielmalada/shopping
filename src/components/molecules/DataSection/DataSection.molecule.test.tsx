import React from 'react';
import renderer from 'react-test-renderer';
import DataSection from './DataSection.molecule';

describe('Data Section', () => {
  it('should be able to render without crashing', () => {
    const tree = renderer.create(<DataSection />).toJSON();

    expect(tree).toMatchSnapshot();
  });
 
});