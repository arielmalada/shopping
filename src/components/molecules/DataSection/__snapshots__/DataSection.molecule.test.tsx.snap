// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Section should be able to render without crashing 1`] = `
<div
  className="space-y-3"
>
  <h2
    className="block antialiased tracking-normal font-sans text-4xl font-semibold leading-[1.3] text-inherit"
  />
  <div>
    <div
      className="flex space-x-2"
    >
      <div>
        $ 
      </div>
    </div>
  </div>
  <div />
</div>
`;
